Service/shipLoader.php

```diff
@@ -47,22 +47,34 @@ public function findOneById($id)
         return $this->createShipFromData($shipArray);
     }
 
-    public function saveShip($shipData)
+    public function saveShip(AbstractShip $shipData)
     {
         $this->shipStorage->saveShipData($shipData);
     }
 
+    public function updateShip(AbstractShip $shipData)
+    {
+        $this->shipStorage->updateShipData($shipData);
+    }
+    // pass in ship obj and grab the ID
+    public function deleteShip($id)
```

Comment:
Pass in the ship object here: `deleteShip(AbstractShip $ship)`

Reply:
Updated